core/src/main/java/greencity/service/impl/TipsAndTricksServiceImpl.java

```diff
@@ -0,0 +1,167 @@
+package greencity.service.impl;
+
+import greencity.constant.CacheConstants;
+import greencity.constant.ErrorMessage;
+import greencity.dto.PageableDto;
+import greencity.dto.tipsandtricks.AddTipsAndTricksDtoRequest;
+import greencity.dto.tipsandtricks.AddTipsAndTricksDtoResponse;
+import greencity.dto.tipsandtricks.TipsAndTricksDto;
+import greencity.entity.TipsAndTricks;
+import greencity.exception.exceptions.NotFoundException;
+import greencity.exception.exceptions.NotSavedException;
+import greencity.repository.TipsAndTricksRepo;
+import greencity.service.FileService;
+import greencity.service.NewsSubscriberService;
+import greencity.service.TipsAndTricksService;
+import greencity.service.TipsAndTricksTagsService;
+import greencity.service.UserService;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.imageio.ImageIO;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import static org.apache.commons.codec.binary.Base64.decodeBase64;
+
+@Service
+@RequiredArgsConstructor
+public class TipsAndTricksServiceImpl implements TipsAndTricksService {
```

Comment:
You need to annotate class with @EnableCaching, otherwise no caching will be performed.

Reply:
Added missing annotation as it was adviced